bootstrap/pkg/kfapp/gcp/gcp.go

```diff
@@ -430,6 +537,9 @@ func (gcp *Gcp) updateDM(resources kftypes.ResourceEnum) error {
 	if err := cred_cmd.Run(); err != nil {
 		return fmt.Errorf("Error when running gcloud container clusters get-credentials: %v", err)
 	}
+	if _, err = os.Stat(kftypes.KubeConfigPath()); !os.IsNotExist(err) {
+		gcp.AddNamedContext()
+	}
```

Comment:
Can we call it at the end of apply()?
Since this function is ```updateDM()```?

Reply:
in `kfctl.sh` we do it in `updateDM` tho: https://github.com/kubeflow/kubeflow/blob/80baee129657f7aa4e93e3adbb820e0629ddbbfd/scripts/gke/util.sh#L158

my guess is because we want to create KUBECONFIG entries immediately after the deployment is created.  let me know if you think moving to `Apply()` is more appropriate, tho.

Reply:
Calling it at the end of apply() won't be conflict with "create KUBECONFIG entries immediately after the deployment is created".
Would be nice to scope down updateDM() to deal with deployment manager logics.

Reply:
Done.